### ISSUE TITLE
Error from 00:00 on the next day

### ISSUE THREAD

**Issue by handjiny** [schnittstabil/csrf-tokenservice#1]
Thank you for publishing a good library.

But there is one problem.

I am using csrf-tokenservice as ssl login from master server to another server.

It is used well without any problem, and error occurs after 00:00 on the next day.

What is the problem? plz help me

**Comment by schnittstabil** [comment]
That's weird. `csrf-tokenservice` uses [`time()`](http://php.net/manual/en/function.time.php) as default value:

https://github.com/schnittstabil/csrf-tokenservice/blob/3b071ecd8e2ddaf84713fa3eeb857004520c4ae7/src/TokenValidator.php#L94-L105

Do you use a non-default `$now` value?

**Comment by handjiny** [comment]
`$now` is null and uses the default value. After the day, I will make a detailed error history after debugging. Thank you for answer

**Comment by handjiny** [comment]
Hi!
I finally found the problem.
The error message is shown below.

Cannot handle token prior to iat: 1504143674, now: 1504143673

I do not know why the request is one second faster.
More strange is that if you try again after a while, it is normal.
I just want to tolerate a one second error.

**Comment by schnittstabil** [comment]
Mhm, do you use a load balancer? Maybe, your servers are out of sync.

Anyway, do you think a `$leeway` parameter would solve your issue? E.g.:

```php
if (abs($now - $payload->iat) < $leeway) { …
```

**Comment by handjiny** [comment]
yes i use load balancer.

I have not had any issues yet.

We will be watching for another few days.  :)

I have not modified the core source because of the update patch

```
$now = time() + 2;
$tokenService->validate($_SERVER['PHP_AUTH_USER'], $token, $now);
```

**Comment by schnittstabil** [comment]
`$now = time() + xxx` solves `exp`, but not `iat` issues :worried: 

I've added a `$leeway` parameter, and have released [v3.1.0](https://github.com/schnittstabil/csrf-tokenservice/releases/tag/3.1.0):

```php
// your workaround:
$tokenService->validate($_SERVER['PHP_AUTH_USER'], $token, time(), 2);

// which is the same as:
$tokenService->validate($_SERVER['PHP_AUTH_USER'], $token, null, 2);
```

**Comment by handjiny** [comment]
Wow very thank you.

You are a really good developer.

Let's check it after the update.
